src/Domains/Inventory/Warehouses/Services/WarehouseService.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Kanvas\Inventory\Warehouses\Services;
+
+use Baka\Users\Contracts\UserInterface;
+use Kanvas\Companies\Models\Companies;
+use Kanvas\Companies\Repositories\CompaniesRepository;
+use Kanvas\Inventory\Status\Models\Status;
+use Kanvas\Inventory\Status\Repositories\StatusRepository;
+use Kanvas\Inventory\Variants\Actions\AddToWarehouseAction as AddToWarehouse;
+use Kanvas\Inventory\Variants\Actions\UpdateToWarehouseAction;
+use Kanvas\Inventory\Variants\DataTransferObject\VariantsWarehouses;
+use Kanvas\Inventory\Variants\Models\Variants;
+use Kanvas\Inventory\Variants\Models\VariantsWarehouses as ModelsVariantsWarehouses;
+use Kanvas\Inventory\Warehouses\Models\Warehouses;
+use Kanvas\Inventory\Warehouses\Repositories\WarehouseRepository;
+
+class WarehouseService
+{
+    /**
+     * Update data of variant in a warehouse.
+     */
+    public static function updateWarehouseVariant(Variants $variant, UserInterface $user, array $warehouses): Variants
+    {
+        $warehousesId = array_column($warehouses, 'warehouse_id');
+
+        $existedWarehouses = $variant->variantWarehouses->keyBy('warehouses_id');
+        $existedWarehouses = $existedWarehouses->keys()->all();
+        $toDelete = array_diff($existedWarehouses, $warehousesId);
+
+        foreach ($warehouses as $warehouseData) {
+            $warehouseModel = WarehouseRepository::getById((int) $warehouseData['warehouse_id'], $variant->product->company()->get()->first());
+
+            if (isset($warehouseData['status'])) {
+                $warehouseData['status_id'] = StatusRepository::getById(
+                    (int) $warehouseData['status']['id'],
+                    $variant->product->company()->get()->first()
+                )->getId();
+            } else {
+                $warehouseData['status_id'] = Status::getDefault($variant->product->company()->get()->first())->getId();
```

Comment:
@arfenis you can do `$variant->product->company` instead of `$variant->product->company()->get()->first()`